packages/cells/src/cells/date-picker-cell.tsx

```diff
@@ -1,15 +1,87 @@
-import * as React from "react";
-import { CustomCell, CustomRenderer, drawTextCell, GridCellKind } from "@glideapps/glide-data-grid";
+import React from "react";
+import {
+    CustomCell,
+    CustomRenderer,
+    drawTextCell,
+    GridCellKind,
+    ProvideEditorCallback,
+} from "@glideapps/glide-data-grid";
 
 interface DatePickerCellProps {
     readonly kind: "date-picker-cell";
     readonly date: Date | undefined;
     readonly displayDate: string;
-    readonly format: "date" | "datetime-local";
+    readonly format: "date" | "time" | "datetime-local";
 }
 
+export const formatValueForHTMLInput = (format: "date" | "time" | "datetime-local", date: Date): string => {
+    if (format === "date") {
+        return date.toISOString().split("T")[0];
+    }
+    if (format === "time") {
+        return date.toISOString().split("T")[1].replace("Z", "");
+    }
+    if (format === "datetime-local") {
+        return date.toISOString().replace("Z", "");
+    }
+    return "";
+};
+
 export type DatePickerCell = CustomCell<DatePickerCellProps>;
 
+const Editor: ReturnType<ProvideEditorCallback<DatePickerCell>> = cell => {
+    const cellData = cell.value.data;
+    const { date, displayDate, format } = cellData;
+    const cellDataDate =
+        cellData.date ??
+        // now as date with adjusted timezone to allow for displayDate to be correct
+        new Date(new Date().getTime() - new Date().getTimezoneOffset() * 60000);
+    const value = formatValueForHTMLInput(format, cellDataDate);
+
+    return (
+        <input
+            required
+            style={{ minHeight: 26, border: "none", outline: "none" }}
+            type={format}
+            value={value}
+            autoFocus={true}
+            onChange={event => {
+                // handle when clear is clicked and value has been wiped
+                if (event.target.value === "") {
+                    try {
+                        cell.onChange({
+                            ...cell.value,
+                            data: {
+                                ...cell.value.data,
+                                // attempt to reset to cached date
+                                date: date !== undefined ? date : new Date(displayDate),
+                            },
+                        });
+                    } catch (error) {
+                        cell.onChange({
+                            ...cell.value,
+                            data: {
+                                ...cell.value.data,
+                                displayDate: String(error),
+                            },
+                        });
+                    }
+                    return;
```

Comment:
I find this early return hurts readability in this case. The error case here is kind of "the bigger chunk" of the method. Can we move this to just a regular if/else setup for now?

Reply:
Are you suggesting this:
```
// handle when clear is clicked and value has been wiped
if (event.target.value === "") {
    cell.onChange({
      ...cell.value,
      data: {
        ...cell.value.data,
        // attempt to reset to cached date
        date: date !== undefined ? date : new Date(displayDate),
      },
    });
  } else {
    cell.onChange({
      ...cell.value,
      data: {
        ...cell.value.data,
        // use valueAsNumber because valueAsDate is null for "datetime-local"
        // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#technical_summary
        date: new Date(event.target.valueAsNumber) ?? cellDataDate,
      },
    });
  }
                        ```

Reply:
yes

Reply:
well kind, you removed the try catch in that block, but essentially yes